src/app/system/sync/fileCache.ts

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useState } from "react"
+import { nanoid } from "nanoid/non-secure"
+import {
+  unlink,
+  mkdir,
+  exists,
+  writeFile,
+  DocumentDirectoryPath,
+  readFile,
+  downloadFile,
+} from "react-native-fs"
+
+const PATH_CACHE = DocumentDirectoryPath + "/cache"
+const PATH_CACHE_IMAGES = PATH_CACHE + "/images"
+const PATH_CACHE_DOCUMENTS = PATH_CACHE + "/documents"
+const PATH_CACHE_JSON = PATH_CACHE + "/json"
+
+interface FileProps {
+  filename: string
+  type: "document" | "json" | "image"
+}
+
+interface SaveFileProps extends FileProps {
+  data: string
+}
+
+export const saveFileToCache = async ({ data, filename, type }: SaveFileProps) => {
+  await makeSureCacheIsReady()
+
+  const path = getFilePath({ type, filename })
+
+  try {
+    await writeFile(path, data, "utf8")
+  } catch (error) {
+    console.log("[fileCache] Error saving file:", error)
+  }
+}
+
+export const downloadImageToCache = async (url: string) => {
+  await makeSureCacheIsReady()
+
+  const id = nanoid()
+  await downloadFile({
+    fromUrl: url,
+    toFile: getFilePath({ type: "image", filename: id + ".png" }),
+  })
+  urlMap[url] = id // TODO: delete old file if it exists
+}
+
+export const getFileFromCache = async ({ filename, type }: FileProps) => {
+  console.log("getting cache from store")
+  const path = getFilePath({ type, filename })
+
+  const fileExists = await exists(path)
+
+  if (!fileExists) {
+    console.error("[fileCache] Error retrieving file from cache", path)
```

Comment:
maybe lets do this as a warning, because error will pop the modal up that usually is used for an app crash. if this is not something that the app would need to crash, then warning is the way to go.
```suggestion
    console.warn("[fileCache] Error retrieving file from cache", path)
```